lib/puppet/defaults.rb

```diff
@@ -647,7 +647,7 @@ def self.default_vendormoduledir
       timers. When these timers drift out of sync, agents can be served
       inconsistent catalogs.",
       :hook => proc do |val|
-        if ![0, 'unlimited'].include?(val)
+        unless [0, 'unlimited', Float::INFINITY].include?(val)
```

Comment:
I also took the opportunity to remove the `if not` in favor of `unless`

Reply:
Will val ever be just the String 'unlimited', or will the Puppet::Settings::TTLSetting type always .munge() it to Float::INFINITY?

Reply:
It looks like it is possible for val to be the String 'unlimited', though that shouldn't happen often. I'd like to keep `'unlimited'` in this array

Reply:
That's frightening :) :+1: